client/src/app/components/common/save-as-modal.component.ts

```diff
@@ -43,15 +43,25 @@ import {ModalService} from "../modal";
 `
 })
 export class SaveAsModalComponent {
+
+    /** Path of the source file that should be copied */
+    public filePath: string;
+
+    /** Toggles the cog loader */
     private isCreatingFile: boolean;
+
+    /** Holds the error object that should be shown below the form if exists */
     private error: {[message: string]: string};
+
+    /** New file destination entry form */
     private newFileForm: ControlGroup;
 
     constructor(private formBuilder: FormBuilder,
                 private eventHub: EventHubService,
-                private workspace: WorkspaceService,
                 private modal: ModalService) {
 
+        this.filePath = undefined;
```

Comment:
@ivanbatic @mayacoda Now I get it: 
https://www.typescriptlang.org/play/#src=class%20Greeter%20%7B%0D%0A%20%20%20%20public%20greeting%3A%20string%3B%0D%0A%20%20%20%20constructor()%20%7B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Aclass%20Greeter2%20%7B%0D%0A%20%20%20%20public%20greeting%3A%20string%3B%0D%0A%20%20%20%20constructor()%20%7B%0D%0A%20%20%20%20%20%20%20%20this.greeting%20%3D%20undefined%3B%0D%0A%20%20%20%20%7D%0D%0A%7D%0D%0A%0D%0Alet%20greeter%20%3D%20new%20Greeter()%3B%0D%0Alet%20greeter2%20%3D%20new%20Greeter2()%3B%0D%0A%0D%0Aconsole.log(greeter.greeting%20%3D%3D%3D%20undefined)%3B%0D%0Aconsole.log(greeter2.greeting%20%3D%3D%3D%20undefined)%3B%0D%0A%0D%0Aconsole.log(Object.keys(greeter))%3B%0D%0Aconsole.log(Object.keys(greeter2))%3B%0D%0A%0D%0A